lib/pipeline.js

```diff
@@ -921,9 +921,14 @@ class PipelineModel extends BaseModel {
                 username,
                 scmContext: this.scmContext
             });
+            let permission = {};
 
-            // eslint-disable-next-line no-await-in-loop
-            const permission = await user.getPermissions(this.scmUri);
+            try {
+                // eslint-disable-next-line no-await-in-loop
+                permission = await user.getPermissions(this.scmUri);
+            } catch (err) {
```

Comment:
can we explicitly check for `401` error ? Wondering if this might cause all admins to be wiped out in case git was down or something.

Reply:
`err` object should be an HttpError object ```{"name":"HttpError","status":401,"hea```

Reply:
updated it